isthmus/src/test/java/io/substrait/isthmus/NestedStructQueryTest.java

```diff
@@ -0,0 +1,427 @@
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Test
+  public void testNested0() {
+    final Table table =
+        new AbstractTable() {
+          @Override
+          public RelDataType getRowType(RelDataTypeFactory factory) {
+            var helper = new TypeHelper(factory);
+            return helper.struct2(
+                "x", helper.i32(),
+                "a", helper.i32());
+          }
+        };
+
+    String query =
+        """
+           SELECT
+             "nested"."my_table"."a"
+           FROM
+             "nested"."my_table";
+           """;
+
+    String expectedExpressionText =
+        """
+          selection {
+            direct_reference {
+              struct_field {
+                field: 1 # a
+              }
+            }
+            root_reference: {}
+          }
+        """;
+
+    test(table, query, expectedExpressionText);
+  }
+
+  @Test
+  public void testNested1() {
+    final Table table =
+        new AbstractTable() {
+          @Override
+          public RelDataType getRowType(RelDataTypeFactory factory) {
+            var helper = new TypeHelper(factory);
+            return helper.struct2(
+                "x", helper.i32(),
+                "a", helper.struct("b", helper.i32()));
+          }
+        };
+
+    String query =
+        """
+           SELECT
+             "nested"."my_table"."a"."b"
+           FROM
+             "nested"."my_table";
+           """;
+
+    String expectedExpressionText =
+        """
+          selection {
+            direct_reference {
+              struct_field {
+                field: 0 # b
+                child {
+                  struct_field {
+                    field: 1 # a
+                  }
+                }
+              }
+            }
+            root_reference: {}
+          }
+        """;
+
+    test(table, query, expectedExpressionText);
+  }
+
+  @Test
+  public void testNested2() {
+    final Table table =
+        new AbstractTable() {
+          @Override
+          public RelDataType getRowType(RelDataTypeFactory factory) {
+            var helper = new TypeHelper(factory);
+            return helper.struct("a", helper.struct("b", helper.struct("c", helper.i32())));
+          }
+        };
+
+    String query =
+        """
+           SELECT
+             "nested"."my_table"."a"."b"."c"
+           FROM
+             "nested"."my_table";
+           """;
+
+    String expectedExpressionText =
+        """
+          selection {
+            direct_reference {
+              struct_field {
+                field: 0 # c
+                child {
+                  struct_field {
+                    field: 0 # b
+                    child: {
+                      struct_field {
+                        field: 0 # a
+                      }
+                    }
+                  }
+                }
+              }
+            }
+            root_reference: {}
+          }
+        """;
```

Comment:
As per discussions in Slack, the references chains should start with the outermost reference (i.e. `a`) not the innermost one.

https://github.com/substrait-io/substrait/pull/520